InteractiveHtmlBom/web/render.js

```diff
@@ -483,21 +484,53 @@ function handlePointerLeave(e, layerdict) {
   delete layerdict.pointerStates[e.pointerId];
 }
 
+function resetTransform(layerdict) {
+  layerdict.transform.panx = 0;
+  layerdict.transform.pany = 0;
+  layerdict.transform.zoom = 1;
+  redrawCanvas(layerdict);
+}
+
 function handlePointerUp(e, layerdict) {
+  if (!e.hasOwnProperty("offsetX")) {
+    // The polyfill doesn't set this properly
+    e.offsetX = e.pageX - e.currentTarget.offsetLeft;
+    e.offsetY = e.pageY - e.currentTarget.offsetTop;
+  }
+
   e.preventDefault();
   e.stopPropagation();
 
-  if (e.button == 0 && layerdict.pointerStates.hasOwnProperty(e.pointerId) && layerdict.pointerStates[e.pointerId].distanceTravelled < 10) {
-    // This is just a click
-    handleMouseClick(e, layerdict);
+  if (layerdict.pointerStates.hasOwnProperty(e.pointerId)) {
+    // We haven't necessarily had a pointermove event since the interaction started, so make sure we update this now
+    var ptr = layerdict.pointerStates[e.pointerId];
+    ptr.distanceTravelled += Math.abs(e.offsetX - ptr.lastX);
+    ptr.distanceTravelled += Math.abs(e.offsetY - ptr.lastY);
+  }
+
+  if (e.button == 0 && layerdict.pointerStates.hasOwnProperty(e.pointerId) && layerdict.pointerStates[e.pointerId].distanceTravelled < 10 && Date.now() - layerdict.pointerStates[e.pointerId].downTime <= 500) {
```

Comment:
This line is way too long, please break it up. And reuse the `ptr` from above.

Reply:
Resolved by not checking for the property and reusing ptr.